### PR TITLE
Add styling to publish page

### PR DESCRIPTION
**Pull request by timcowlishaw**
## Changes in this PR:
Adds @Donna-H's styling to @jacksonj04's new publish flow, as well as updating the only edit page to reuse the same layout components, and styling the document review form.

A couple of notes:

 - the 'minor issues checklist' is behind the `log_minor_issues_on_publish` feature flag which will need enabling once the required backend work is complete.
 
 - similarly, the 'assign judgment' option in the publish sidebar is behind the `assign_in_publish_sidebar` feature flag, which, again, will need enabling once any required backend work is completed.
 
- We've also created a reusable button-cta css class for consistency, following the pattern introduced with Terry's button-secondary.

- We've had to remove curlylint from the pre-commit flow, as it chokes on waffle feature flag tags. The plan is to replace with djlint once this is merged!

## Trello card / Rollbar error (etc)
https://trello.com/c/MRWkx0SG/612-%F0%9F%8E%AB-eui-basic-publish-screen
and 
https://trello.com/c/T3QVjOZS/619-%F0%9F%93%99-eui-basic-successfully-published-screen

### Before
Screenshot 2023-03-15 at 14 36 18

### After
Screenshot 2023-03-15 at 14 35 58
Screenshot 2023-03-21 at 13 41 08
Screenshot 2023-03-21 at 13 40 29

### CODE REVIEW

**Comment by timcowlishaw** [comment]
chatted with @Terry-Price @Donna-H  and @jacksonj04 - there's a few outstanding styling tweaks to be made that @Donna-H and I will look at tomorrow.

File: ds_caselaw_editor_ui/templates/judgment/publish-success.html [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-            <p>{% include 'includes/email_confirm_publication.html' %}</p>
-          </div>
-        </div>
-        <aside class="publish-component__sidebar">
-          {% include "includes/publish_sidebar.html" %}
-        </aside>
-      </div>
-    </div>
+  <h2>Success - this document has been published
+    <a href="{{ context.judgment.public_uri }}" target="_blank" rel="noopener noreferrer"><br>
```
Comment (jacksonj04) [inline ds_caselaw_editor_ui/templates/judgment/publish-success.html:13]: I'm not sure this `<br>` semantically belongs within the link, it should probably be on the line above.

File: ds_caselaw_editor_ui/sass/includes/publish.scss
Comment (jacksonj04) [inline ds_caselaw_editor_ui/sass/includes/publish.scss:1]: This is still called publish.scss, is there a better name for this?

File: ds_caselaw_editor_ui/sass/includes/publish.scss [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  @media (max-width: 800px) {
+    .edit-component__actions {
+      display: block;
+    }
+    .create-email-block {
+      margin: 20px 0;
+    }
+  }
+
+.publish-sidebar {
```
Comment (jacksonj04) [inline ds_caselaw_editor_ui/sass/includes/publish.scss:108]: I don't think this class is used any more.

**Review by jacksonj04 — CHANGES_REQUESTED** [review]
Really nice work, especially in the refactoring to clean things up. A couple of code tweaks I've flagged where I think things could be deleted, and a couple of whitespace questions (see image below), but other than that it's good to go!

Screenshot 2023-03-21 at 16 21 18

**Review by jacksonj04 — APPROVED** [review]
🎉

**Pull request merged**